tests/validation/validation_data.cpp

```diff
@@ -15,6 +16,38 @@ namespace mc {
 
 trace_io g_trace_io;
 
+util::path trace_io::find_datadir() {
+    // If environment variable is set, use that in preference.
+    if (const char* env_path = std::getenv("NMC_DATADIR")) {
+        return util::path(env_path);
```

Comment:
It's best-effort plus override behaviour. Where we draw a line around 'best-effort' is certainly debatable, but I think we should defer to the user (i.e. environment variable or command-line parameter) if they provide a path explicitly at run time.